### DISCUSSION TITLE
Profile Page Analytics

### DISCUSSION THREAD

**Discussion by DarrachBarneveld** [codu-code/codu#580] [Ideas]
## Context

It would be cool if thee profile page had some analytics. At least some simple ones to start. You could also reward engaging users with level ups similar to discord.

## Expected Behavior
A simple analytics bar to start could incorporate - 
1. Total posts
2. Engagement metrics based on comments or likes
3. Total favourites (could link to saved posts)
4. Pending posts( link to drafts or posts awaiting approval)
5. Total followers (future feature)

## Current Behavior

The current behaviour on the profile page lists published articles, total amount and a list.

## Screenshots
![Screenshot 2023-10-19 at 22 34 35](https://github.com/codu-code/codu/assets/130994206/1331b1ed-c8a4-4912-b48a-3c72c88f5a02)

## Additional info

Let me know what you think I could whip up a quick prototype

**Comment by NiallJoeMaher** [comment]
I'm on the fence about metrics. Reason being, if something doesn't have a lot of ratings people might not bother going and seeing it. 

It might be useful to have some personal metrics that are private to the user or were you thinking so on the users public feed we could see some of the metrics?

**Comment by DarrachBarneveld** [comment]
Yeah personal metrics, can mostly be private to the user. Currently total articles is displayed publicly but it might be cool to add a similar level system seen in discord to reward active posters. Or micro badges for completing the writing challenge.  

Private data could just be a view where it's easy for a user to see which of their posts where most engaged with, and maybe navigation to their saved posts. Future adaptions you could have a follow system so you would see how many/which users you follow. Basically more profile/user account content info.

Could be very basic to begin with

**Comment by NiallJoeMaher** [comment]
Moving this to discussions so we can start hashing out more ideas 🦾

**Comment by DarrachBarneveld** [comment]
Cool, never used a github discussion before. Exciting. I have some time this evening to make a small prototype so I can visually show what I mean :)

**Reply by DarrachBarneveld** [reply]
Hey sorry was super busy finishing off some stuff so I did a real quick mockup. Its ugly but just for reference 
![ezgif com-video-to-gif](https://github.com/codu-code/codu/assets/130994206/1db2e897-9698-48e8-831f-f288cb8a879b)
